tests/capture/transforms/test_capture_commute_controlled.py

```diff
@@ -0,0 +1,696 @@
+        assert len(jaxpr.eqns) == 1
+        circuit_jaxpr = jaxpr.eqns[0].params["qfunc_jaxpr"]
+        assert len(circuit_jaxpr.eqns) == 16
+
+        result = jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts)
+
+        with qml.capture.pause():
+            # pylint: disable=not-callable
+            expected_result = circuit()
+
+        assert qml.math.allclose(result, expected_result)
+
+    @pytest.mark.parametrize(
+        "selector, expected_ops",
+        [
+            (
+                0.2,
+                [
+                    qml.CNOT(wires=[0, 1]),
+                    qml.CNOT(wires=[0, 2]),
+                    qml.Toffoli(wires=[0, 1, 2]),
+                    qml.RX(np.pi, wires=2),
+                    qml.CNOT(wires=[0, 1]),
+                ],
+            ),
+            (
+                0.4,
+                [
+                    qml.CNOT(wires=[0, 1]),
+                    qml.CNOT(wires=[0, 2]),
+                    qml.Toffoli(wires=[0, 1, 2]),
+                    qml.RX(np.pi, wires=2),
+                    qml.CNOT(wires=[0, 1]),
+                ],
+            ),
+            (
+                0.8,
+                [
+                    qml.CNOT(wires=[0, 1]),
+                    qml.RY(np.pi, wires=2),
+                    qml.CNOT(wires=[0, 2]),
+                    qml.Toffoli(wires=[0, 1, 2]),
+                    qml.CNOT(wires=[0, 1]),
+                ],
+            ),
+        ],
+    )
+    def test_cond(self, selector, expected_ops):
+        """Test that operations inside a conditional block are correctly pushed."""
+
+        @CommuteControlledInterpreter()
+        def circuit(selector, x):
+
+            def true_branch(x):
+                qml.RY(x, wires=2)
+                qml.CNOT(wires=[0, 2])
+                qml.Toffoli(wires=[0, 1, 2])
+
+            # pylint: disable=unused-argument
+            def false_branch(x):
+                qml.RX(x, wires=2)
+                qml.CNOT(wires=[0, 2])
+                qml.Toffoli(wires=[0, 1, 2])
+
+            qml.CNOT(wires=[0, 1])
+            qml.cond(selector > 0.5, true_branch, false_branch)(x)
+            qml.CNOT(wires=[0, 1])
+
+        jaxpr = jax.make_jaxpr(circuit)(selector, np.pi)
+
+        assert len(jaxpr.eqns) == 4
+        assert jaxpr.eqns[0].primitive == jax.lax.gt_p
+        assert jaxpr.eqns[1].primitive == qml.CNOT._primitive
+        assert jaxpr.eqns[2].primitive == cond_prim
+        assert jaxpr.eqns[3].primitive == qml.CNOT._primitive
+
+        collector = CollectOpsandMeas()
+        collector.eval(jaxpr.jaxpr, jaxpr.consts, selector, np.pi)
+        jaxpr_ops = collector.state["ops"]
+
+        for op1, op2 in zip(jaxpr_ops, expected_ops, strict=True):
+            qml.assert_equal(op1, op2)
+
+    def test_for_loop(self):
+        """Test that for operators inside a for loop are correctly pushed."""
+
+        @CommuteControlledInterpreter()
+        def circuit(x):
+
+            @qml.for_loop(0, 1)
```

Comment:
Can we make this iterate at least twice to include boundary behaviour in the test? :) (like in other transform PR)